lib/eunit/src/eunit_lib.erl

```diff
@@ -574,6 +574,17 @@ trie_match([], []) ->
 trie_match([], _T) ->
     no.
 
+%% ---------------------------------------------------------------------
+%% Convert a string of unknown encoding to a binary
+
+try_to_binary(S) when is_list(S) ->
```

Comment:
~s and ~ts always expects lists to contain unicode codepoints. But ~s checks that all code points are below 255.

There is no (or should not be any)  code in otp that expects utf-8 encoded bytes in lists, that is a really bad representation causing confusion.
You should represent strings as lists of unicode codepoints or a binary containing utf-8 encoded bytes or latin-1 encoded bytes.